metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiRecord.java

```diff
@@ -0,0 +1,42 @@
+package eu.europeana.metis.harvesting.oaipmh;
+
+import eu.europeana.metis.harvesting.HarvesterException;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+/**
+ * This is a immutable class representing an (OAI-embedded) record along with it's OAI header.
+ */
+public class OaiRecord {
+
+  private final OaiRecordHeader header;
+  private final byte[] record;
+
+  /**
+   * Constructor.
+   *
+   * @param header The OAI header of the record.
+   * @param record The record itself (the embedded record).
+   */
+  public OaiRecord(OaiRecordHeader header, byte[] record) {
+    this.header = header;
+    this.record = record;
+  }
+
+  public OaiRecordHeader getHeader() {
+    return header;
+  }
+
+  /**
+   * Makes the embedded record available.
+   *
+   * @return An input stream containing the record. The caller needs to close it after use.
+   * @throws HarvesterException In case the record is marked as deleted.
+   */
+  public InputStream getRecord() throws HarvesterException {
```

Comment:
I was thinking, is there a way to protect the system by somehow auto closing the InputStream if the caller didn't. 
Even in the tests they are not closed I think, not that it matters that much..
E.g. holding a map of items for the class and close them at the end or something like that? Or is it an overkill, what do you think?

Reply:
Indeed the byte stream is safer! I guess we can stick to the javadoc notification